src/main/java/com/solace/samples/java/snippets/HowToImplementTracingManualInstrumentation.java

```diff
@@ -0,0 +1,182 @@
+                .setAttribute(SemanticAttributes.MESSAGING_TEMP_DESTINATION, false)
+                // Set more attributes as needed
+                //.setAttribute(...)
+                //.setAttribute(...)
+                .setParent(Context.current()) // set current context as parent
+                .startSpan();
+
+        // Create baggage object
+        Baggage baggage = BaggageUtil.extractBaggage("key1=val1,key2=val2");
+
+        // set sendSpan as new current context
+        try (Scope scope = sendSpan.makeCurrent()) {
+            // set baggage in the current context
+            try (Scope scope1 = baggage.storeInContext(Context.current()).makeCurrent()) {
+                final SolacePubSubPlusJavaTextMapSetter setter = new SolacePubSubPlusJavaTextMapSetter();
+                final TextMapPropagator propagator = openTelemetry.getPropagators().getTextMapPropagator();
+                // then inject current context with send span and baggage into the message
+                propagator.inject(Context.current(), message, setter);
+                // publish message to the given topic
+                messagePublisher.publish("simple message to the world", messageDestination);
+            }
+        } catch (Exception e) {
+            sendSpan.recordException(e); // Span can record exception if any
+            sendSpan.setStatus(StatusCode.ERROR, e.getMessage()); // Set span status as ERROR/FAILED
+        } finally {
+            sendSpan.end(); // End sendSpan. Span data is exported when span.end() is called.
+        }
+    }
+
+    /**
+     * Example how to extract a tracing context from the Solace PubSub+ Java Message and generate a RECEIVE span
+     * for the received message.
+     *
+     * @param receivedMessage  A Solace PubSub+ Java Message.
+     * @param messageProcessor A callback function that the user could use to process a message
+     * @param openTelemetry    The OpenTelemetry class is the entry point to telemetry functionality
+     *                         for tracing, metrics and baggage from OpenTelemetry Java SDK.
+     * @param tracer           OpenTelemetry Tracer is the interface from OpenTelemetry Java SDK for
+     *                         span creation and interaction with the in-process context.
+     */
+    void howToCreateNewSpanOnMessageReceive(InboundMessage receivedMessage,
+                                            Consumer<InboundMessage> messageProcessor,
+                                            OpenTelemetry openTelemetry, Tracer tracer) {
+
+        // Extract tracing context from message, if any using the SolacePubSubPlusJavaTextMapGetter
+        final SolacePubSubPlusJavaTextMapGetter getter = new SolacePubSubPlusJavaTextMapGetter();
+        final Context extractedContext = openTelemetry.getPropagators().getTextMapPropagator()
+                .extract(Context.current(), receivedMessage, getter);
```

Comment:
What is the point of the `getter` object?  Can you use it again?  If not, no need to define it?  Or maybe add some comment/description of what a "SolacePubSubPlusJavaTextMapGetter" (what a name!) object is.